extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,26 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
```

Comment:
$product will need to be detainted here.
IMHO it's probably safe enough to untaint() if it's not a ref, e.g.
```perl
if (!ref $params->{product}) {
   untaint($params->{product});
}
else {
   ThrowUserError(TODO); # we have an error name for bad parameters, but I forget what it is
}
```